CMakeLists.txt

```diff
@@ -93,6 +93,18 @@ if(GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
     if(NOT GIT_SUBMOD_RESULT EQUAL "0")
         message(FATAL_ERROR "Cloning git submodules failed with ${GIT_SUBMOD_RESULT}, please checkout submodules manually")
     endif()
+    # Apply patches; check if patch is already applied first, and apply it if not
+    execute_process(COMMAND ${GIT_EXECUTABLE} -C tpls/pybind11/ apply ../customizations/pybind11/pybind.h.diff --reverse --check
```

Comment:
I think that doing it this way will leave developers (working in local clones) with a permanent unstaged change from simply running `cmake` to configure / build the project. This is not what we want. You could instead just add pybind as a proper git submodule, make the "docs fix" change there in `tpls/pybind11` and commit it, and then `git add tpls/pybind11`.

Reply:
pybind11 is a proper submodule, but I don't want to fork it. Rather than forking or undoing the patch at the end, configuring the parent repo to not list the submodule change I think is the cleanest option. You will still be able to work with the submodule normally.